Source/Details/ASGraphicsContext.mm

```diff
@@ -78,10 +78,25 @@ NS_INLINE void ASConfigureExtendedRange(UIGraphicsImageRendererFormat *format)
         ASConfigureExtendedRange(format);
       }
 
-      return [[[UIGraphicsImageRenderer alloc] initWithSize:size format:format] imageWithActions:^(UIGraphicsImageRendererContext *rendererContext) {
-        ASDisplayNodeCAssert(UIGraphicsGetCurrentContext(), @"Should have a context!");
-        work();
-      }];
+      // Avoid using the imageWithActions: method because it does not support cancellation at the
+      // last moment i.e. before actually creating the resulting image.
+      __block UIImage *image;
+      NSError *error;
+      [[[UIGraphicsImageRenderer alloc] initWithSize:size format:format]
```

Comment:
`ASGraphicsCreateImageWithOptions` may return `nil` in case of error or cancelation. It should be reflected in header.

Reply:
Agree.